src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -650,7 +650,7 @@ function renderSubtreeIntoContainer(
 
 var ReactDOMFiber = {
   hydrate(
-    element: ReactElement<any>,
+    element: React$Element<any>,
```

Comment:
Ditto. (And actually this should take React.Node I think).

Reply:
To be honest, I'm still a little unclear on when we should be using one vs the other, but I think you're right that the [`Node` type](https://flow.org/en/docs/react/types/#toc-react-node) would work here.